interview-questions.md

```diff
@@ -8,33 +8,64 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+(source Language): JSX (JavaScript + XML) is an extension of JavaScript that permits you to write down HTML directly within JavaScript. which features a few benefits of creating your code more readable and exercising the complete power of JavaScript within HTML. JSX is in some ways almost like HTML, however, it does accompany certain conspicuous differences which we’ll cover within the next section. Since JSX isn’t a legitimate JS code, it must be compiled into JS with a tool like Babel or similar.
+
+(My Interpretation)
+JSX is an extension of javascript(I know the source said that) that filled a need to brige javascript and HTML for better functionality when programing. And like many things where there a benefits there are also shortcomings in the way of having to only return a single parent element. 
 
 1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
-Your answer:
+Your answer: Yarn is Yet Another Resource Negotiator. the files that are modified when runnin yarn are the node module files to add dependencies for related files and the yarn lock file that is created.
 
 Researched answer:
 
+(The Source: https://en.wikipedia.org/wiki/Yarn_(package_manager))
+
+(language from the source: Yarn is one of the main JavaScript package managers,developed in 2016 by Meta (formerly Facebook) for the Node.js JavaScript runtime environment. An alternative to the npm package manager, Yarn was created as a collaboration of Facebook, Exponent (now Expo.dev), Google, and Tilde (the company behind Ember.js) to solve consistency, security, and performance problems with large codebases.)
+
+(My Interpretation: Yarn as it's name states is Yet Another Resource negotiator. It is a package manager that will find, download and install files that you need for certain task. Looking into what yarn is ultimatily wasnt as frutifull in helping me understand as looking into what a package manager is. Because that is ultimatly what yarn is.)
+
 3. What is a React component?
 
-Your answer:
+Your answer: A react component is part if an overall react build. Best prectices seems to be to separate your compnentents for their specific purpose to prevent corrumption of the overall build.
 
-Researched answer:
+Researched answer: 
+(the source: https://www.w3schools.com/react/react_components.asp#:~:text=Components%20are%20independent%20and%20reusable,will%20concentrate%20on%20Function%20components.)(secondary source: Manager XD Development Team at Illumina)
+
+(language from the source: Components are independent and reusable bits of code. They serve the same purpose as JavaScript functions, but work in isolation and return HTML.
+Components come in two types, Class components and Function components, in this tutorial we will concentrate on Function components.)
+
+(My interpretation: Components are are part of the build that are segrigated to make the code more D.R.Y and are resubale and make the code easier to read)
 
```

Comment:
what is more content that you can add to your answers to show you have experience building React components